### PR TITLE
WT-6592 Avoid marking errors for skipped Python tests due to not-built extension

### PR DESCRIPTION
**Pull request by lukech**
Introducing a new `skipped` instance variable in `WiredTigerTestCase` Python class. Setting the variable while detecting not-built extension and calling skipTest(), and at the tearDown phase avoid treating the skipped test as an error.

Please note normally when a test is marked as skipped in unittest framework, no setUp() or tearDown() would be called. The case we are trying to address here is a bit special, that the test is not skipped at the beginning (as a whole), instead, the skip decision is made at a later stage of the test when detecting missing extension.

### CODE REVIEW

File: test/suite/wttest.py [hunk 1]
```diff
@@ -283,6 +284,7 @@ def extensionsConfig(self):
             filenames = glob.glob(pat)
             if len(filenames) == 0:
                 if skipIfMissing:
+                    self.skipped = True
```
Comment (ddanderson) [inline test/suite/wttest.py:287]: If code in other places calls `self.skipTest` (for example `test_bulk01.py`), the `self.skipped` variable won't be set.  We would like to set it in `self.skipTest` itself, but I see that it is defined in a class we don't own.

I think the solution is to create method `WiredTigerTestCase.skipTest(self)`, and set the variable in it, and call `unittest.TestCase.skipTest(self)`.  It's a little more complicated, but I think it's worth getting right for the general case.

File: test/suite/wttest.py [hunk 2]
```diff
@@ -457,17 +459,18 @@ def tearDown(self):
             for f in files:
                 os.chmod(os.path.join(root, f), 0o666)
         self.pr('passed=' + str(passed))
+        self.pr('skipped=' + str(self.skipped))
 
         # Clean up unless there's a failure
-        if passed and not WiredTigerTestCase._preserveFiles:
+        if passed and not WiredTigerTestCase._preserveFiles or self.skipped:
```
Comment (ddanderson) [inline test/suite/wttest.py:465]: The code is correct, but I would add parentheses around the `and` part of the expression.   I never fully remember the order, and every time I see the code, I have to think about it.  Adding parentheses reduces the cognitive load.

**Review by ddanderson — COMMENTED** [review]
The fix is correct, but I think it can be improved.  I've noted a couple items.

File: test/suite/wttest.py (hunk 1 above)
Comment (lukech) [reply test/suite/wttest.py]: Thanks for the good suggestion. I've implemented the method solution as suggested in cbc9142.

File: test/suite/wttest.py (hunk 2 above)
Comment (lukech) [reply test/suite/wttest.py]: Readability counts, reworked!

**Review by ddanderson — APPROVED** [review]
LGTM!

**Pull request merged**